hal/src/hal-filesystem.ads

```diff
@@ -0,0 +1,145 @@
+with Interfaces; use Interfaces;
+
+package HAL.Filesystem is
+
+   subtype Pathname is String;
+
+   type File_Kind is (Regular_File, Directory);
+
+   type File_Mode is (Read_Only, Write_Only, Read_Write);
+
+   type Permission_Kind  is (Others_Execute, Others_Write, Others_Read,
```

Comment:
I don't agree that access control is overkill for the Ada_Drivers_Library, it's key feature of file systems, even if older ones don't implement it. But I don't mind leaving that out for the moment.